dev/src/platform.cpp

```diff
@@ -568,32 +568,36 @@ void SetConsole(bool on) {
     #endif
 }
 
-iint LaunchSubProcess(const char **cmdl, const char *stdins, string &out) {
+iint LaunchSubProcess(const char** cmdl, const char* stdins, string& out) {
     #ifndef PLATFORM_ES3
         struct subprocess_s subprocess;
-        int result = subprocess_create(cmdl, subprocess_option_inherit_environment, &subprocess);
+        int result = subprocess_create(cmdl,
+            subprocess_option_inherit_environment
+            | subprocess_option_enable_async, &subprocess);
         if (result) return -1;
         if (stdins) {
-            FILE *p_stdin = subprocess_stdin(&subprocess);
+            FILE* p_stdin = subprocess_stdin(&subprocess);
             fputs(stdins, p_stdin);
             fclose(p_stdin);
         }
+        const unsigned int buflen = 256;
+        const char buf[buflen] = "";
+        unsigned int readlength = subprocess_read_stdout(&subprocess, (char* const)buf, buflen);
+        while (readlength) {
```

Comment:
why have same code twice? refactor into `for(;;) { .. if (..) break; .. }` loop.

Reply:
the old code had a lambda to do the same 2 identical loops, why not retain that?

Reply:
I am not entirely sure how to make a lambda in this case - I would have to pass in the functions themselves as an argument?

Reply:
Should I use std::bind?

Reply:
Just use the same code as was there originally?

Reply:
look at the code you replaced.

Reply:
So it would look something like `unsigned int (*subprocess_read)(subprocess_s *const, char *const, const unsigned int)`